If the packages are signed, and the signature is not in the rpm-native
RPMDB, an error will occur.  We want to avoid this failure mode 
when building the native version of createrepo.

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

Index: createrepo-0.4.11/dumpMetadata.py
===================================================================
--- createrepo-0.4.11.orig/dumpMetadata.py
+++ createrepo-0.4.11/dumpMetadata.py
@@ -92,7 +92,7 @@ def returnHdr(ts, package):
             fdno = package # let's assume this is an fdno and go with it :)
     except OSError:
         raise MDError, "Error opening file"
-    ts.setVSFlags((rpm.RPMVSF_NOMD5|rpm.RPMVSF_NEEDPAYLOAD))
+    ts.setVSFlags((rpm.RPMVSF_NOMD5|rpm.RPMVSF_NEEDPAYLOAD|rpm.RPMVSF_NODSA|rpm.RPMVSF_NORSA|rpm.RPMVSF_NODSAHEADER|rpm.RPMVSF_NORSAHEADER))
     try:
         hdr = ts.hdrFromFdno(fdno)
     except rpm.error:
